src/main/java/io/syndesis/qe/marketplace/manifests/Index.java

```diff
@@ -0,0 +1,129 @@
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+import io.fabric8.openshift.client.OpenShiftClient;
+import io.syndesis.qe.marketplace.openshift.OpenShiftService;
+import io.syndesis.qe.marketplace.quay.QuayService;
+import io.syndesis.qe.marketplace.quay.QuayUser;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Predicate;
+
+import static io.syndesis.qe.marketplace.util.HelperFunctions.readResource;
+import static io.syndesis.qe.marketplace.util.HelperFunctions.waitFor;
+
+@Slf4j
+public class Index {
+    private final String name;
+    @Getter
+    private String ocpName;
+    private List<Bundle> bundles;
+    private Opm opm;
+    private boolean isPushed;
+    private static final String BUILD_TOOL = "docker";
+    static final String MARKETPLACE_NAMESPACE = "openshift-marketplace";
+    private static QuayService quaySvc;
+
+    Index(String name) {
+        this.name = name;
+        bundles = new ArrayList<>();
+        isPushed = false;
+    }
+
+    void pull(QuayUser user) {
+        cmdWrapper(BUILD_TOOL, "login", "-u=" + user.getUserName(), "-p=" + user.getPassword(), "quay.io");
+        int ret = cmdWrapper(BUILD_TOOL, "pull", this.name);
+        if (ret != 0) {
+            throw new IllegalStateException("Could not pull " + this.name + ", please verify it is pushed");
+        }
+
+    }
+
+    void setOpm(Opm opm) {
+        this.opm = opm;
+    }
+
+    public Bundle addBundle(String bundleName) {
+        int ret = opm.runCmd("index", "add", "--bundles=" + bundleName, "--tag=" + this.name, "--build-tool=" + BUILD_TOOL);
+        isPushed = false;
+        Bundle bundle = new Bundle(bundleName, this);
+        bundles.add(bundle);
+        return bundle;
+    }
+
+    public void addBundles(String... names) {
+        for (String name : names) {
+            addBundle(name);
+        }
+    }
+
+    @SneakyThrows
+    private int cmdWrapper(String... command) {
+        Process process = new ProcessBuilder(command).start();
+        int ret = process.waitFor();
+        if (ret != 0) {
+            String out = IOUtils.toString(process.getErrorStream(), Charset.defaultCharset());
+            log.error(out);
+        }
+        return ret;
+    }
+
+    @SneakyThrows
+    public void push(QuayUser user) {
+        if (quaySvc == null) {
+            quaySvc = new QuayService(user, null, null);
+        }
+        cmdWrapper(BUILD_TOOL, "login", "-u=" + user.getUserName(), "-p=" + user.getPassword(), "quay.io");
+        cmdWrapper(BUILD_TOOL, "push", name);
+        String[] parts = name.split("/");
+        String name = parts[parts.length - 1].replaceAll(":.*", "");
```

Comment:
it is just me, or is this not used? :grin:

Reply:
this has to be used from the testsuite code

Reply:
I meant this one variable that seem to be not used :grin:

Reply:
:man_facepalming: